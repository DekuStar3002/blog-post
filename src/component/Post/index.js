import React, { useState, useEffect } from "react";
import "./Post.css";
import { API_ENDPOINT } from "../../constants";
import MakeRequest from "../../utils/makeRequest";
import blackClapIcon from "../../assets/Icons/clapping-black.svg";
import greyClapIcon from "../../assets/Icons/clapping-grey.svg";
import redLikeIcon from "../../assets/Icons/heart-red.svg";
import blackLikeIcon from "../../assets/Icons/heart-black.svg";
import { PostImage, PostContent } from "..";

function Post({
  index,
  id,
  date,
  readingTime,
  title,
  description,
  claps,
  liked,
  image,
  handleClap,
  handleLike,
}) {
  return (
    <div className="post" data-testid='posts'>
      <PostImage imgSrc={image} />
      <PostContent
        index={index}
        id={id}
        date={date}
        readingTime={readingTime}
        title={title}
        description={description}
        clapIcon={blackClapIcon}
        claps={claps}
        likeIcon={liked ? redLikeIcon : blackLikeIcon}
        handleClap={handleClap}
        handleLike={handleLike}
      />
    </div>
  );
}

export default Post;
