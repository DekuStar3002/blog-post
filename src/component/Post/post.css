.post {
  height: 60vh;
  width: 20vw;
  border: 1px solid black;
}

.img {
  height: 60%;
  width: 100%;
}

.img img {
  height: 100%;
  width: 100%;
}

.content {
  margin: 1vh 1vw;
}

.upper {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.title {
  font-size: x-large;
}

.middle {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  margin: 2vh 0;
}

.bottom {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 1vh 0;
}

.clap {
  display: flex;
  align-items: center;
  justify-content: center;
}

.reaction {
  height: 3vh;
  width: 3vw;
}