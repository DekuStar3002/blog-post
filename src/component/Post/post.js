import React from 'react';
import './Post.css';
import clapIcon from '../../assets/Icons/clapping.svg';
import redLikeIcon from '../../assets/Icons/heart-red.svg';
import blackLikeIcon from '../../assets/Icons/heart-black.svg';
const Post = ({date, readingTime, title, description, claps, liked, image}) => {
  const imgSrc = require(`../../assets/Images/${image}`);
  return (
    <div className='post'>
      <div className='img'>
        <img src={imgSrc} alt='imgSrc'/>
      </div>
      <div className='content'>
        <div className='upper'>
          <p>{date}</p>
          <p>{readingTime}</p>
        </div>
        <div className='middle'>
          <p className='title'>{title}</p>
          <p className='about'>{description}</p>
        </div>
        <hr />
        <div className='bottom'>
          <div className='clap'>
            <img className='reaction' src={clapIcon} alt='clapping'/>
            <p>{claps}</p>
          </div>
          <img className='reaction' src={liked ? redLikeIcon : blackLikeIcon}  alt='like'/>
        </div>
      </div>
    </div>
  );
}

export default Post;