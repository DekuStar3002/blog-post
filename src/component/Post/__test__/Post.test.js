import Post from '..';
import { render } from '@testing-library/react';

describe('Post', () => { 
  it('should render correctly', () => {
    const props = {
      id: 1, 
      date: 'date', 
      readingTime: 'readingTime', 
      title: 'title', 
      description: 'description', 
      claps: 5, 
      claped: false, 
      clickClap: jest.fn(), 
      liked: false, 
      clickLike: jest.fn(), 
      image: 'abstract.png'
    }
    const { asFragment } = render(<Post {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  })
});