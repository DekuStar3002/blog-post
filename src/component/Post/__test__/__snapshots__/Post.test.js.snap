// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly 1`] = `
<DocumentFragment>
  <div
    class="post"
    data-testid="posts"
  >
    <div
      class="img"
    >
      <img
        alt="imgSrc"
        src="abstract.png"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="upper"
      >
        <p>
          NaNth
      undefined, 
      NaN
        </p>
        <p>
          readingTime
        </p>
      </div>
      <div
        class="middle"
      >
        <p
          class="title"
        >
          title
        </p>
        <p
          class="about"
        >
          description
        </p>
      </div>
      <hr
        class="horizontalLine"
      />
      <div
        class="bottom"
      >
        <div
          class="clap"
        >
          <img
            alt="clapping"
            class="reaction"
            data-testid="clap-src"
            src="clapping-black.svg"
          />
          <p
            data-testid="claps"
          >
            5
          </p>
        </div>
        <img
          alt="like"
          class="reaction heart"
          data-testid="heart"
          src="heart-black.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
